src/transports/tcp.rs

```diff
@@ -437,6 +437,63 @@ impl<Id: PeerId> Transport<Id> for TcpTransport<Id> {
         Ok(())
     }
 
+    fn send_timeout(
+        endpoint: &mut TcpEndpoint,
+        data: &[u8],
+        timeout: Duration,
+    ) -> Result<(), crate::error::PeerNetErrorData> {
+        let start_time = std::time::Instant::now();
+
+        let msg_size: u32 = data.len().try_into().map_err(|_| {
+            TcpError::ConnectionError
+                .wrap()
+                .error("send len too long", Some(format!("{:?}", data.len())))
+        })?;
+        if msg_size > 1048576000 {
+            return Err(TcpError::ConnectionError
+                .wrap()
+                .error("send len too long", Some(format!("{:?}", data.len()))))?;
+        }
+        //TODO: Use config one
+
+        let stream = &mut endpoint.stream.stream;
```

Comment:
Why do you use the underline socket directly ? And why did you made the loop above ?

Reply:
> Why do you use the underline socket directly ? And why did you made the loop above ?

`endpoint.stream` is now `Limiter<TcpStream>` and if i want the `TcpStream` i need to made endpoint.stream.stream  .

maybe we should rename the Limiter to `limiter` instead of `stream` ?

The while loop ? (because timeout does not work with write_all)

Reply:
Why you can't use the limiter directly ?

Ok for the loop I understand now

Reply:
The Limiter does not have `set_write_timeout`  :)

Reply:
Ok can you set the write_timeout on the stream directly but still the limiter to write and avoid writing too much chunks at once ?

Reply:
The first version implemented with write_all here : https://github.com/massalabs/PeerNet/pull/50/files/b9680291bca3e79a9a4896df1fe9bc84863f01b6#diff-9b5647e5e8b0377c67e4af6bc5437f81ccb342a869e041223fbea66487b96a89

I think i have to add after send, this :
```rust 
  stream.set_write_timeout(None).map_err(|e| {
            PeerNetError::SendError.error("error set write timeout", Some(e.to_string()))
        })?;

```
For remove the set_write_timeout on the stream, what do you think @AurelienFT

Reply:
I think we misunderstand each other. I asked if it's possible that still use the underline stream to set the timeout as you made but to use the wrapped limited stream to `write` in your loop ?

Reply:
> I think we misunderstand each other. I asked if it's possible that still use the underline stream to set the timeout as you made but to use the wrapped limited stream to `write` in your loop ?

i understand, no problem with that. I will make it.

For `set_write_timeout(None)` it is just another question. I think he should be reset.